src/components/Footer/Footer.js

```diff
@@ -53,23 +76,99 @@ const Footer = ({
     }
     return '#C4C4C4';
   };
+
+  const username = item?.anon_user_info_emoji_name
+    ? `Anonymous ${item?.anon_user_info_emoji_name}`
+    : item?.actor?.data?.username;
+
+  const onPressDM = async () => {
+    try {
+      setLoading({...loading, loadingDm: true});
+      if (!item?.anon_user_info_emoji_name) {
+        const channelName = [username, profile?.myProfile?.username].join(',');
+        const selectedUser = {
+          user: {
+            name: channelName,
+            image: item?.actor?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+          }
+        };
+        const members = [item?.actor?.id, profile.myProfile.user_id];
+        await createSignChat(members, selectedUser);
+      } else {
+        await sendMessageDM(item.id, 'post', 'SIGNED');
+      }
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDm: false});
+    }
+  };
+
+  const onPressDMAnon = async () => {
+    try {
+      setLoading({...loading, loadingDmAnon: true});
+      await sendMessageDM(item.id, 'post', 'ANONYMOUS');
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDmAnon: false});
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This section introduces two new functions: `onPressDM` and `onPressDMAnon`. These functions handle sending direct messages, either signed or anonymous. Error handling is done using try-catch blocks, which is good. However, the error logging could be improved. Instead of just warning the error, consider implementing a more user-friendly way to display errors, such as a toast notification or an error message in the UI.



<!-- This is an auto-generated comment by OSS CodeRabbit -->